service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/DraftCMOController.java

```diff
@@ -0,0 +1,61 @@
+package uk.gov.hmcts.reform.fpl.controllers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.swagger.annotations.Api;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.hmcts.reform.ccd.client.model.AboutToStartOrSubmitCallbackResponse;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.model.CaseManagementOrder;
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicList;
+import uk.gov.hmcts.reform.fpl.service.DraftCMOService;
+
+import java.util.Map;
+
+@Api
+@RestController
+@RequestMapping("/callback/draft-cmo")
+public class DraftCMOController {
+    private final ObjectMapper mapper;
+    private final DraftCMOService draftCMOService;
+
+    @Autowired
+    public DraftCMOController(ObjectMapper mapper, DraftCMOService draftCMOService) {
+        this.mapper = mapper;
+        this.draftCMOService = draftCMOService;
+    }
+
+    @PostMapping("/about-to-start")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackrequest) {
+        CaseDetails caseDetails = callbackrequest.getCaseDetails();
+        Map<String, Object> caseData = caseDetails.getData();
+
+        caseData.put("cmoHearingDateList", draftCMOService.getHearingDateDynamicList(caseDetails));
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseData)
+            .build();
+    }
+
+    @PostMapping("/about-to-submit")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        DynamicList list = mapper.convertValue(caseDetails.getData().get("cmoHearingDateList"), DynamicList.class);
+
+        CaseManagementOrder caseManagementOrder = CaseManagementOrder.builder()
+            .hearingDate(list.getValue().getLabel())
+            .hearingDateId(list.getValue().getCode())
+            .build();
```

Comment:
this should probably happen in the service. I got a bit lost just looking for it... new method that creates a CMO and takes in the value of the dynamic list and returns a CMO object

Reply:
Moved to service